### PR TITLE
[ML][Transforms] unifying logging, adding some more logging

### PR DESCRIPTION
**Pull request by benwtrent**
This PR unifies our logging messages in the transform task so that they are easier to grok. Some new logging entries are added, but one logging message is removed (it was a hold over from previous debugging efforts). 

Additionally, the "when to audit on checkpoint finish" logic is now changed so that we write an audit message for nicely rounded numbers such as `10`, `100`, and not `19`, `99` or `199`.

### CODE REVIEW

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java [hunk 1]
```diff
@@ -304,6 +303,7 @@ public synchronized void start(Long startingCheckpoint, boolean force, ActionLis
                 listener.onResponse(new StartDataFrameTransformTaskAction.Response(true));
             },
             exc -> {
+                logger.error("[" + transform.getId() + "] failed updating state to [" + state + "].", exc);
```
Comment (przemekwitek) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java]: Is it possible to use interpolation markers ("{}") here (and in other places) the same way as you did in line 355?

Comment (benwtrent) [reply x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java]: Its not possible to use interpolation and have an exception too.

**Comment by benwtrent** [comment]
run elasticsearch-ci/1

**Comment by benwtrent** [comment]
@elasticmachine update branch

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java]: A, ok. Understand.

Comment (davidkyle) [reply x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java]: Use a `ParameterizedMessage` 

```suggestion
                logger.error(new ParameterizedMessage("[{}] failed updating state to [{}].", transform.getId(), state), exc);
```

Comment (przemekwitek) [reply x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java]: +1
I forgot about it. Seems like a perfect solution to maintain consistency of how the messages are crafted in this file.

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java [hunk 2]
```diff
@@ -1049,7 +1070,9 @@ private boolean isIrrecoverableFailure(Exception e) {
         }
 
         synchronized void handleFailure(Exception e) {
-            logger.warn("Data frame transform [" + transformTask.getTransformId() + "] encountered an exception: ", e);
+            logger.warn(new ParameterizedMessage("[{} data frame transform encountered an exception: ",
```
Comment (przemekwitek) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java]: Missing ']'

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java [hunk 3]
```diff
@@ -923,7 +932,9 @@ protected void onFailure(Exception exc) {
                 }
                 handleFailure(exc);
             } catch (Exception e) {
-                logger.error("Data frame transform encountered an unexpected internal exception: " ,e);
+                logger.error(
+                    new ParameterizedMessage("[{} data frame transform encountered an unexpected internal exception: ", transformId),
```
Comment (przemekwitek) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java]: Missing ']'

**Review by przemekwitek — APPROVED** [review]
Two minor comments, otherwise LGTM

**Pull request merged**